File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    internal int ExpectedMinutesInOven() => 40;

    internal int RemainingMinutesInOven(int elapsedMinutes) => ExpectedMinutesInOven() - elapsedMinutes;

    internal int PreparationTimeInMinutes(int layers) => layers * 2;

    internal int ElapsedTimeInMinutes(int layers, int elapsedMinutes) => elapsedMinutes + PreparationTimeInMinutes(layers);
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven elapsedMinutes =
    expectedMinutesInOven - elapsedMinutes

let preparationTimeInMinutes layers =
    layers * 2

let elapsedTimeInMinutes layers elapsedMinutes =
    preparationTimeInMinutes layers + elapsedMinutes
